Build fails when PRINTING is not defined

Undefining `PRINTING` in `osqp_configure.h` results in the following build error: 

```
../../rt/3rdparty/osqp/src/util.c: In function ‘copy_settings’:
../../rt/3rdparty/osqp/src/util.c:227:3: error: implicit declaration of function ‘memcpy’ [-Werror=implicit-function-declaration]
   memcpy(new, settings, sizeof(OSQPSettings));
   ^
../../rt/3rdparty/osqp/src/util.c:227:3: error: incompatible implicit declaration of built-in function ‘memcpy’ [-Werror]
../../rt/3rdparty/osqp/src/util.c:227:3: note: include ‘<string.h>’ or provide a declaration of ‘memcpy’
cc1: all warnings being treated as errors
```

This happens because `<string.h>` is only included when `PRINTING` is defined. 

This isn't actually a problem for me (I just ran into it while testing out build flags), but I figured it was worth mentioning.

--- a/ci/appveyor/script.cmd
+++ b/ci/appveyor/script.cmd
@@ -33,4 +33,17 @@ cmake --build .
 %APPVEYOR_BUILD_FOLDER%\build\out\osqp_tester.exe
 if errorlevel 1 exit /b 1
 
+
+
+:: Perform C Tests without printing
+:: -----------------------------------------------------
+cd %APPVEYOR_BUILD_FOLDER%
+rmdir /s /q build
+mkdir build
+cd build
+cmake -G "%CMAKE_PROJECT%" -DPRINTING=OFF -DUNITTESTS=ON ..
+cmake --build .
+%APPVEYOR_BUILD_FOLDER%\build\out\osqp_tester.exe
+if errorlevel 1 exit /b 1
+
 @echo off

--- a/ci/travis/script.sh
+++ b/ci/travis/script.sh
@@ -69,3 +69,14 @@ cd build
 cmake -G "Unix Makefiles" -DDLONG=OFF -DUNITTESTS=ON ..
 make
 ${TRAVIS_BUILD_DIR}/build/out/osqp_tester
+
+
+
+echo "Testing OSQP without printing"
+cd ${TRAVIS_BUILD_DIR}
+rm -rf build
+mkdir build
+cd build
+cmake -G "Unix Makefiles" -DPRINTING=OFF -DUNITTESTS=ON ..
+make
+${TRAVIS_BUILD_DIR}/build/out/osqp_tester

--- a/include/auxil.h
+++ b/include/auxil.h
@@ -101,7 +101,7 @@ c_float compute_obj_val(OSQPWorkspace *work,
 
 /**
  * Check whether QP has solution
- * @param work Workspace
+ * @param info OSQPInfo
  */
 c_int has_solution(OSQPInfo *info);
 

--- a/include/auxil.h
+++ b/include/auxil.h
@@ -99,6 +99,11 @@ void update_y(OSQPWorkspace *work);
 c_float compute_obj_val(OSQPWorkspace *work,
                         c_float       *x);
 
+/**
+ * Check whether QP has solution
+ * @param work Workspace
+ */
+c_int has_solution(OSQPInfo *info);
 
 /**
  * Store the QP solution

--- a/lin_sys/lib_handler.c
+++ b/lin_sys/lib_handler.c
@@ -2,6 +2,7 @@
 #include <ctype.h> // Needed for tolower functions
 
 #include "constants.h"
+#include "util.h"
 
 soHandle_t lh_load_lib(const char *libName) {
     soHandle_t h = OSQP_NULL;
@@ -110,7 +111,7 @@ symtype lh_load_sym (soHandle_t h, const char *symName) {
             tripSym = ucbuf;
             break;
         case 4:                             /* original_ */
-            memcpy (ocbuf, symName, symLen);
+            c_strcpy(ocbuf, symName);
             ocbuf[symLen] = '_';
             ocbuf[symLen+1] = '\0';
             tripSym = ocbuf;

--- a/src/auxil.c
+++ b/src/auxil.c
@@ -516,7 +516,7 @@ c_int is_dual_infeasible(OSQPWorkspace *work, c_float eps_dual_inf) {
   return 0;
 }
 
-static c_int has_solution(OSQPInfo * info){
+c_int has_solution(OSQPInfo * info){
 
   return ((info->status_val != OSQP_PRIMAL_INFEASIBLE) &&
       (info->status_val != OSQP_PRIMAL_INFEASIBLE_INACCURATE) &&

--- a/src/osqp.c
+++ b/src/osqp.c
@@ -508,7 +508,6 @@ c_int osqp_solve(OSQPWorkspace *work) {
 
   }        // End of ADMM for loop
 
-
   // Update information and check termination condition if it hasn't been done
   // during last iteration (max_iter reached or check_termination disabled)
   if (!can_check_termination) {
@@ -534,13 +533,15 @@ c_int osqp_solve(OSQPWorkspace *work) {
 
     /* Check whether a termination criterion is triggered */
     check_termination(work, 0);
+
   }
 
   // Compute objective value in case it was not
   // computed during the iterations
-  if (!compute_cost_function) {
+  if (!compute_cost_function && has_solution(work->info)){
     work->info->obj_val = compute_obj_val(work, work->x);
   }
+
 
 #ifdef PRINTING
   /* Print summary for last iteration */
@@ -552,7 +553,6 @@ c_int osqp_solve(OSQPWorkspace *work) {
   /* if max iterations reached, change status accordingly */
   if (work->info->status_val == OSQP_UNSOLVED) {
     if (!check_termination(work, 1)) { // Try to check for approximate
-                                       // termination
       update_status(work->info, OSQP_MAX_ITER_REACHED);
     }
   }
@@ -983,7 +983,7 @@ c_int osqp_update_P(OSQPWorkspace *work,
   c_int i;        // For indexing
   c_int exitflag; // Exit flag
   c_int nnzP;     // Number of nonzeros in P
-  
+
 #ifdef PROFILING
   if (work->clear_update_time == 1) {
     work->clear_update_time = 0;

--- a/src/util.c
+++ b/src/util.c
@@ -224,7 +224,38 @@ OSQPSettings* copy_settings(OSQPSettings *settings) {
   OSQPSettings *new = c_malloc(sizeof(OSQPSettings));
 
   // Copy settings
-  memcpy(new, settings, sizeof(OSQPSettings));
+  // NB. Copying them explicitly because memcpy is not
+  // defined when PRINTING is disabled (appears in string.h)
+  new->rho = settings->rho;
+  new->sigma = settings->sigma;
+  new->scaling = settings->scaling;
+
+# if EMBEDDED != 1
+  new->adaptive_rho = settings->adaptive_rho;
+  new->adaptive_rho_interval = settings->adaptive_rho_interval;
+  new->adaptive_rho_tolerance = settings->adaptive_rho_tolerance;
+# ifndef PROFILING
+  new->adaptive_rho_fraction = settings->adaptive_rho_fraction;
+# endif
+# endif // EMBEDDED != 1
+  new->max_iter = settings->max_iter;
+  new->eps_abs = settings->eps_abs;
+  new->eps_rel = settings->eps_rel;
+  new->eps_prim_inf = settings->eps_prim_inf;
+  new->eps_dual_inf = settings->eps_dual_inf;
+  new->alpha = settings->alpha;
+  new->linsys_solver = settings->linsys_solver;
+  new->delta = settings->delta;
+  new->polish = settings->polish;
+  new->polish_refine_iter = settings->polish_refine_iter;
+  new->verbose = settings->verbose;
+  new->scaled_termination = settings->scaled_termination;
+  new->check_termination = settings->check_termination;
+  new->warm_start = settings->warm_start;
+# ifdef PROFILING
+  new->time_limit = settings->time_limit;
+# endif
+
   return new;
 }
 

--- a/tests/non_cvx/test_non_cvx.h
+++ b/tests/non_cvx/test_non_cvx.h
@@ -53,7 +53,7 @@ static char* test_non_cvx_solve()
 
   // Compare objective values
   mu_assert("Non Convex test solve: Error in objective value!",
-            work->info->obj_val == OSQP_NAN);
+	    work->info->obj_val == OSQP_NAN);
 
   // Clean workspace
   osqp_cleanup(work);

